sql/src/main/java/io/crate/operation/user/User.java

```diff
@@ -76,6 +76,18 @@ public boolean hasPrivilege(Privilege.Type type, Privilege.Clazz clazz, String i
         return hasPrivilege.isPresent();
     }
 
+    /**
+     * Checks if the user has any privilege that matches the given class, type and ident
+     * currently we check for any privilege, since Class is always CLUSTER and ident null.
+     *
+     * @param clazz privilege class (ie. CLUSTER, TABLE, etc)
+     * @param ident ident of the object
+     */
+    public boolean hasAnyPrivilege(Privilege.Clazz clazz, String ident) {
+        Optional<Privilege> hasPrivilege = privileges.stream().findAny();
```

Comment:
We are creating so many objects here. privileges.size > 0 should do the trick too